Fix 'unused variable' warning in the code generated by CodeGen_C with 'no_asserts' feature

vksnk opened the pull request:
Tested with HL_TARGET=host-no_asserts make -C apps/c_backend/ test

abadams commented:
I think the create_assertion that takes two strings is:

1) Going to have the same problem
2) Implies you've evaluated the message and put the result in a variable, even though that would be the wrong semantics
3) Is only used in one place, which could be changed to pass a Call Expr for the message

So it should be deleted.

vksnk commented:
> I think the create_assertion that takes two strings is:
> 
> 1. Going to have the same problem
> 2. Implies you've evaluated the message and put the result in a variable, even though that would be the wrong semantics
> 3. Is only used in one place, which could be changed to pass a Call Expr for the message
> 
> So it should be deleted.

Thanks for the suggestion! I missed it, because in this specific case assert was checking that the memory allocation is not nullptr, so there was no warning for unused memory, because the pointer was actually used later in the program.

I've removed create_assertion(string, string) and replaced it's only occurrence with Call, PTAL.
